### ISSUE TITLE
Microsoft Defender has detected this file as a virus

### ISSUE THREAD

**Issue by AoiNiCo** [34736384/genshin-fps-unlock#294]
Microsoft Defender has detected this file as a virus
![image](https://github.com/34736384/genshin-fps-unlock/assets/52157900/1132791b-0e17-46ee-92cf-67fa09485d15)

**Comment by rhonivanpareja** [comment]
Yeah i have recieved the same notification on my antivirus, can someone review the changes why the file is being detected?

**Comment by Jonyee04** [comment]
it seems i am not alone, i got the same virus type, strange because yesterday i played without any warning

**Comment by Jonyee04** [comment]
i already removed because of security 
![imagem_2023-12-21_140016530](https://github.com/34736384/genshin-fps-unlock/assets/103516099/6b534126-b725-4694-948d-e322e010bfa7)

**Comment by Jonyee04** [comment]
also, we got a response for the file creator earlier on his own post
![imagem_2023-12-21_140343279](https://github.com/34736384/genshin-fps-unlock/assets/103516099/cacc8ffe-cd87-46fd-a276-cb96f0d1e048)

**Comment by tuanminh26596** [comment]
I know the software creator has put a lot of effort and dedication into maintaining and updating the software until now. But the current version is having a lot of problems with Windows Defender, especially when it classifies this software into many different types of viruses.
![image](https://github.com/34736384/genshin-fps-unlock/assets/60099719/30448ed6-7b78-4170-8978-0f07c9a9f222)

**Comment by Jonyee04** [comment]
@tuanminh26596 yeah i do appreciate as well, it's nice to have someone like him supporting the comunity, but the only reason i don't let any file that gets a warning in my system it's because i suffered a lot with trojans back then, i think it's a valid point

**Comment by Clavus** [comment]
I've tried building the project myself and Windows Defender still tripped over the executable Visual Studio spat out. So yeah, it's some virus scanner heuristic that's resulting in a false positive.

I'm not much of a C++ dev myself otherwise it'd be interesting to dive in to see if there's a workaround to avoid the trigger.

**Comment by tuanminh26596** [comment]
> @tuanminh26596 yeah i appreciate it too, it's nice to have someone like him supporting the comunity, but the only reason i don't let any file that gets a warning in my system it's because i suffered a lot with trojans back then, i think it's a valid point

Have to play Genshin at 60 fps on a 2k 164hz monitor is a torture [💀](https://emojipedia.org/skull)

**Comment by Andrejob99** [comment]
yes, please fix

**Comment by EdwinStoteler** [comment]
Looking at the PR to 4.3 update I expect this line in the vcxproj file to be the issue:
`<UACExecutionLevel>RequireAdministrator</UACExecutionLevel>`

At least I don't see any other chances that the program wasn't doing already before. Perhaps someone could make a branch and test it. Make a PR if it works. I don't have the tools installed on my PC (not going to use my work laptop for this).

**Comment by MenonPlayz** [comment]
The downloadable .exe doesnt get detected by VirusTotal, neither by Malwarebytes.
![grafik](https://github.com/34736384/genshin-fps-unlock/assets/90968284/4a802368-8efc-409c-98b8-7f7b07260c62)

> I also built it myself and it worked for some reason. Microsoft isn't detecting it. 
I tried the same, downloaded Visual Studio 2022, built the source code and scanned it with Windows Defender and Malwarebytes. No detection. Am using the self-built now but I wouldn't worry too much

**Comment by Perceval21** [comment]
Porbably the problem is that Win Def is really picky.

**Comment by Rabydex556** [comment]
Hello. 7 hours ago today I received the same error. I tried to reinstall everything and I still get the Windows Defender alert. Hopefully the creator of an explanation or a patch to solve this quickly. I want to enjoy the new event at 120fps :(

**Comment by Delpirou-Maxime** [comment]
i have the same "issue" but as the dev said, it's a false positive. 

You  have the same issue with cheatengine for example (windows defender and some other Anti Virus said it's a trojan/malware) but if you download from a trust source there's no problem.

But again you can built in yourself but if i've read correctly even this trigger some Anti-Virus. 


I choose to trust the dev.

**Comment by MenonPlayz** [comment]
Okay, I built everything since the May 29 commit ["try to fix unlocker not working on first boot"](https://github.com/34736384/genshin-fps-unlock/commit/07e9c9449847cde4a5347ec343b7ef69b07cc3e3?diff=unified&w=0)

The VirusTotal links for my self compiled [May 29](https://www.virustotal.com/gui/file/3e15ec24fcee913e282d657a9994355cd721b96f2280ec8b116e86bc0c8533e0), [November 17 commit](https://www.virustotal.com/gui/file/b9b1d214cb67904e10b1e7597a57b168e1a8fd326ab32410b2da8e05878779b7), [December 6](https://www.virustotal.com/gui/file/bf61d49f33dbca2c53592011c39f368292a153b75f2a3c3718780ce9f8294bd7), [December 7](https://www.virustotal.com/gui/file/4b3e9804d724352f5d5922890ac93c297ae9cebaa61b8908d8752e31055c150d), [December 16](https://www.virustotal.com/gui/file/906b52cd50166d36aefec44af71a7637cbdf2bfd19eb3ed46a3b8ef4d75fd2d7), [December 17 - 1](https://www.virustotal.com/gui/file/8422ab1498cadb767b668f6bf01d606baa225627cb40c829496265764e069996), [December 17 - 2](https://www.virustotal.com/gui/file/30cefcb137f031d515268fb6180d51d17b2dd6df8a2fac767ba4201d1bfe36db) and [December 19 aka latest commit](https://www.virustotal.com/gui/file/e0e36bce6796a0e9d51966957d0945f7b895ed7782e3ffa734f5da1496741322) builds seem to show the same detections.
They are different from scanning the downloaded exe of the 2.0.0 release, which gives [this VirusTotal report](https://www.virustotal.com/gui/file/40515ad6b9444753609a2c2d7c4da5855796dbdb202f0a0888f9c7e64800772c).

None get detected by Microsoft Defender (both on my PC and in VirusTotal, I manually selected and scanned all of the exes).
While doing this, I realised I should check older releases as a comparison. 
[Release v1.4.2](https://www.virustotal.com/gui/file/b40b9fff8d7472e46454323df8cbbabf0b30aa07cd04b5560056e7b03da72b48) and [Release v1.0.0](https://www.virustotal.com/gui/file/f62fae37db7503ea3c98c81f87d1d87b811579872190b97f4134b33606ed853a) did show some slightly different things.

I dont know a lot about malware, am new to C++ and new to this unlocker. So I can only give my findings to y'all

**Comment by Rabydex556** [comment]
> Bien, construí todo desde la confirmación del 29 de mayo ["intente arreglar que el desbloqueo no funcione en el primer arranque"](https://github.com/34736384/genshin-fps-unlock/commit/07e9c9449847cde4a5347ec343b7ef69b07cc3e3?diff=unified&w=0)
> 
> Los enlaces de VirusTotal para mi autocompilación [del 29 de mayo](https://www.virustotal.com/gui/file/3e15ec24fcee913e282d657a9994355cd721b96f2280ec8b116e86bc0c8533e0) , [confirmación del 17 de noviembre](https://www.virustotal.com/gui/file/b9b1d214cb67904e10b1e7597a57b168e1a8fd326ab32410b2da8e05878779b7) , [6 de diciembre](https://www.virustotal.com/gui/file/bf61d49f33dbca2c53592011c39f368292a153b75f2a3c3718780ce9f8294bd7) , [7 de](https://www.virustotal.com/gui/file/4b3e9804d724352f5d5922890ac93c297ae9cebaa61b8908d8752e31055c150d) diciembre , [16 de diciembre](https://www.virustotal.com/gui/file/906b52cd50166d36aefec44af71a7637cbdf2bfd19eb3ed46a3b8ef4d75fd2d7) , [17 y 1 de diciembre](https://www.virustotal.com/gui/file/8422ab1498cadb767b668f6bf01d606baa225627cb40c829496265764e069996) , [17 y 2 de diciembre](https://www.virustotal.com/gui/file/30cefcb137f031d515268fb6180d51d17b2dd6df8a2fac767ba4201d1bfe36db) y [19 de diciembre, también conocidas como las últimas](https://www.virustotal.com/gui/file/e0e36bce6796a0e9d51966957d0945f7b895ed7782e3ffa734f5da1496741322) compilaciones de confirmación, parecen mostrar las mismas detecciones. Son diferentes de escanear el archivo ejecutable descargado de la versión 2.0.0, que proporciona [este informe de VirusTotal](https://www.virustotal.com/gui/file/40515ad6b9444753609a2c2d7c4da5855796dbdb202f0a0888f9c7e64800772c) .
> 
> Microsoft Defender no detecta ninguno (tanto en mi PC como en VirusTotal, seleccioné y escaneé manualmente todos los archivos ejecutables). Mientras hacía esto, me di cuenta de que debería comprobar las versiones anteriores a modo de comparación. [La versión v1.4.2](https://www.virustotal.com/gui/file/b40b9fff8d7472e46454323df8cbbabf0b30aa07cd04b5560056e7b03da72b48) y [la versión v1.0.0](https://www.virustotal.com/gui/file/f62fae37db7503ea3c98c81f87d1d87b811579872190b97f4134b33606ed853a) mostraron algunas cosas ligeramente diferentes.
> 
> No sé mucho sobre malware, soy nuevo en C++ y nuevo en este desbloqueador. Así que sólo puedo darles mis hallazgos a todos ustedes.

Excuse me, can you give me a small class on how to compile from Visual Studio so I can use it in official Genshin? I am 100% new to programming
